src/pages/index.js

```diff
@@ -30,55 +36,198 @@ const addCardPopup = new PopupWithForm(
+
+  api
+    .patchProfileInfo(name, about)
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+    });
 }
 
 function handleAddCardSubmit(newCardData) {
-  renderer(newCardData);
-  addCardPopup.close();
+  addCardPopup.renderLoading(true);
+
+  api
+    .postCards(newCardData)
+    .then((cardData) => {
+      renderer(cardData);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+}
+
+function handleAvatarSubmit(formData) {
+  const avatarLink = formData["modal__form-input-link"];
+
+  editAvatarPopup.renderLoading(true);
+
+  api
+    .patchProfileAvatar(avatarLink)
+    .then(() => {
+      userInfo.setAvatarPic(avatarLink);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoading(false);
+    });
+}
+
+function handleImageClick(card) {
+  previewImagePopup.open({ name: card.name, link: card.link });
+}
+
+function handleDelete(card) {
+  trashConfirmPopup.open();
+  trashModalSubmitBtn.addEventListener(
+    "click",
+    () => {
+      api
+        .deleteCard(card._id)
+        .then(() => {
+          card.removeCard();
+          trashConfirmPopup.close();
+        })
+        .catch((err) => {
+          console.error("Error deleting card:", err);
+        });
+    },
+    { once: true }
+  );
 }
 
-function handleImageClick(name, link) {
-  previewImagePopup.open({ name, link });
+function handleLikeIconClick(card) {
+  const apiAction = card._isLiked
+    ? api.deleteCardLike(card._id)
+    : api.putCardLike(card._id);
+
+  apiAction
+    .then((updatedCard) => {
+      card.updateHeartIcon();
+    })
+    .catch((err) => {
+      console.error(
+        `Error ${card._isLiked ? "unliking" : "liking"} card:`,
+        err
+      );
+    });
 }
 
-// Renderer + Cards
-function createCard(item) {
-  const card = new Card(item, "#card__template", handleImageClick);
-  return card.getView();
+function createCard(cardData) {
+  const cardInstance = new Card(
+    {
+      name: cardData.name,
+      altName: cardData.altName || "",
+      link: cardData.link,
+      _id: cardData._id,
+      isLiked: cardData.isLiked || false,
+      likes: cardData.likes,
+    },
+    "#card__template",
+    handleImageClick,
+    () => handleDelete(cardInstance),
+    () => handleLikeIconClick(cardInstance)
+  );
+
+  return cardInstance.getView();
 }
 
-function renderer(item) {
-  const cardElement = createCard(item);
+function renderer(cardData) {
+  const cardElement = createCard(cardData);
   cardSection.addItem(cardElement);
 }
+
+// Initialize Section
+cardSection.renderItems();
```

Comment:
`cardSection.renderItems();
`

It's not needed anymore because you should get the cards from the server first and only then call `renderItems`